system/Model.php

```diff
@@ -97,14 +97,14 @@ class Model extends BaseModel
 	 */
 	public function __construct(ConnectionInterface &$db = null, ValidationInterface $validation = null)
 	{
-		parent::__construct($validation);
-
 		/**
 		 * @var BaseConnection $db
 		 */
 		$db = $db ?? Database::connect($this->DBGroup);
 
 		$this->db = &$db;
+
+		parent::__construct($validation);
```

Comment:
This can be reverted to its original place.

Reply:
I think this needs to stay last. Even your example above requires it here:

> For example if the parent initializes the $db property and I'll gonna be using that later for setting up a db prefix

Otherwise `initialize()` won't have access to `$this->db`. Call stack should look like:
1. `Model::__construct()`
2. `BaseModel::__construct()`
3. `MyModel::initialize()`

That way the developer gets the very last line in the stack so has the most properties available.